pennylane/qaoa/mixers.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This file contains built-in functions for constructing QAOA mixer Hamiltonians.
+"""
+import networkx as nx
+import pennylane as qml
+from pennylane.wires import Wires
+
+
+def x_mixer(wires):
+    r"""Creates a basic Pauli-X mixer Hamiltonian.
+
+    This Hamiltonian is defined as:
+
+    .. math:: H_M \ = \ \displaystyle\sum_{i} X_{i},
+
+    where :math:`i` ranges over all wires, and :math:`X_i`
+    denotes the Pauli-X operator on the :math:`i`-th wire.
+
+    This is mixer is used in *A Quantum Approximate Optimization Algorithm*
+    by Edward Farhi, Jeffrey Goldstone, Sam Gutmann [`arXiv:1411.4028 <https://arxiv.org/abs/1411.4028>`__].
+
+
+    Args:
+        wires (Iterable or Wires): The wires on which the Hamiltonian is applied
+
```

Comment:
For both mixers, let's add a code snippet showing usage details. See for example https://pennylane.readthedocs.io/en/latest/code/api/pennylane.templates.layers.SimplifiedTwoDesign.html to get an idea of how this can be done. In this case, can be short and sweet, maybe a couple of lines